### PR TITLE
Flip Report to Skip_Report for postsubmit jobs

### PR DESCRIPTION
**Pull request by krzyzacy**
Works for https://github.com/kubernetes/test-infra/commits/master (click the green check-mark from each merge commits, you can see the context and detail link etc)

Also dry'd out SkipReport and Context && Added an announcement.

(We only have ~10s of postsubmits, should not have negative impacts on our token usage)

/assign @cjwagner @fejta @stevekuznetsov

### CODE REVIEW

**Comment by krzyzacy** [comment]
(doh - will fix tests tomorrow)

File: prow/config/jobs.go [hunk 1]
```diff
@@ -148,12 +147,8 @@ type Postsubmit struct {
 
 	Brancher
 
-	// Context is the name of the GitHub status context for the job.
-	Context string `json:"context"`
-
-	// TODO(krzyzacy): opt-in for now - Consider make it default true like presubmits
-	// Report will comment and set status on GitHub.
-	Report bool `json:"report,omitempty"`
+	// TODO(krzyzacy): Move existing `Report` into `Skip_Report` once this is deployed
```
Comment (stevekuznetsov) [inline prow/config/jobs.go]: TODO not needed anymore?

**Comment by krzyzacy** [comment]
@fejta It gives useful information on commit status for a repo and will also be straight forward to check the branch health from the repo/commit page, thus I think is a useful thing to have by default for all prow users.

**Comment by krzyzacy** [comment]
I'm amused by the number of different kind of `jab`, `jeb`, `jib`, `job`, `jub` we have :laughing:

File: prow/config/jobs.go (hunk 1 above)
Comment (stevekuznetsov) [reply prow/config/jobs.go]: :man_facepalming:

**Comment by stevekuznetsov** [comment]
`jib` `jab` `jub` are awful for debugging

**Comment by stevekuznetsov** [comment]
@fejta why should reporting behavior be different for different types of jobs?

**Comment by krzyzacy** [comment]
> `jib` `jab` `jub` are awful for debugging

luckily we only have five vowels :-)

File: prow/ANNOUNCEMENTS.md [hunk 2]
```diff
@@ -1,7 +1,9 @@
 # Announcements
 
 New features added to each component:
-
+ - *February 6, 2019* prow (both plank and crier) will set status on the commit
+   for postsubmit jobs on github. You can add `Skip_Report: true` to opt-out of
```
Comment (fejta) [inline prow/ANNOUNCEMENTS.md]: isn't it `skip_report: true`?

Comment (krzyzacy) [reply prow/ANNOUNCEMENTS.md]: yep, done

**Comment by krzyzacy** [comment]
oh well...
/pony rebase

**Comment by fejta** [comment]
Can we put the behavior for whether or not postsubmits report by default behind a flag (that defaults to off, which we can set to true in the future?)

**Comment by krzyzacy** [comment]
> Can we put the behavior for whether or not postsubmits report by default behind a flag (that defaults to off, which we can set to true in the future?)

I probably can add a config field

**Comment by krzyzacy** [comment]
Doh - I was planning to centralize report control in the reporter interface for crier but looks like there are many ways to report to github :disappointed: I think explicitly list out job types we want to report is the closest idea I can think of...

**Comment by krzyzacy** [comment]
added more tests, PTAL at the last commit

File: prow/ANNOUNCEMENTS.md [hunk 3]
```diff
@@ -1,7 +1,11 @@
 # Announcements
 
 New features added to each component:
-
+ - *February 13, 2019* prow (both plank and crier) can set status on the commit
+   for postsubmit jobs on github now! 
+   Type of jobs can be reported to github is gated by `github_reporter: job_types_to_report`
```
Comment (fejta) [inline prow/ANNOUNCEMENTS.md]: `github_reporter: ["presubmit", "postsubmit"]`, right?

File: prow/config/config.go [hunk 4]
```diff
@@ -732,6 +743,18 @@ func parseProwConfig(c *Config) error {
 		c.Gerrit.RateLimit = 5
 	}
 
+	if len(c.GithubReporter.JobTypesToReport) == 0 {
+		c.GithubReporter.JobTypesToReport = append(c.GithubReporter.JobTypesToReport, prowapi.PresubmitJob)
```
Comment (fejta) [inline prow/config/config.go]: Is this default we always want? If not I would suggest adding a TODO to update this so we don't forget

File: prow/ANNOUNCEMENTS.md (hunk 3 above)
Comment (krzyzacy) [reply prow/ANNOUNCEMENTS.md]: pasted a real blob here

File: prow/config/config.go (hunk 4 above)
Comment (krzyzacy) [reply prow/config/config.go]: will switch to presubmit + postsubmit, added a TODO

File: prow/ANNOUNCEMENTS.md [hunk 5]
```diff
@@ -1,7 +1,19 @@
 # Announcements
 
 New features added to each component:
-
+ - *February 13, 2019* prow (both plank and crier) can set status on the commit
+   for postsubmit jobs on github now! 
+   Type of jobs can be reported to github is gated by a config field like
+   ```
```
Comment (fejta) [inline prow/ANNOUNCEMENTS.md]: ` ```yaml` will give you syntax highlighting

**Pull request merged**